src/core/network/qgsblockingnetworkrequest.cpp

```diff
@@ -256,7 +256,7 @@ QgsBlockingNetworkRequest::ErrorCode QgsBlockingNetworkRequest::doRequest( QgsBl
       if ( !threadFinished )
       {
         waitConditionMutex.unlock();
-
+        QThread::currentThread()->usleep( 100 );
         QgsApplication::instance()->processEvents();
```

Comment:
Why not waiting for the handleRequest to be taken care of with an eventloop like this 
```
QEventLoop loop;
connect ( QgsNetworkAccessManager::instance(), &QgsNetworkAccessManager::requestAuthDetailsAdded, &loop, &QEventLoop::quit );
loop.exec();
```

I understand with the PR #7055 that there is issues with QEventLoop in main thread, do you know why ?